MarkBind/markbind
# Add Annotation Feature

**What is the purpose of this pull request?**


Fixes #898

**Overview of changes:**

*Note that the implementation has changed completely after discussion with @damithc 

Added basic annotation feature into markbind with the idea of allowing users to add popovers and text to their existing images. Now works by making use of the existing floating-vue lib to help manage popovers over the image. Users will now specify the image to annotate then add child `<a-points>` to draw circles over the image which can function as markers for annotations. 

Example Code: 
```
<annotate src="https://www.researchgate.net/profile/Chaiwat-Sakul/publication/228949121/figure/fig2/AS:300654661783574@1448693065750/The-previous-square-root-circuit-1.png" alt="pic" eager type="point" width="500" >
  <a-point x="25%" y="30%" body="Lorem ipsum dolor sit amet" />
  <a-point x="50%" y="50%" label="1" size="50" header="Lorem ipsum"  body="Lorem ipsum dolor sit amet" legend="bottom" color="blue" opacity="0.1" size="100" />
</annotate>
```

Example Output:
![image](https://user-images.githubusercontent.com/19247856/163362979-cfeac321-385d-44bf-9e96-cc8212cd08d8.png)

The first `<a-point>` is the minimal template where an x, y and text body has to be entered. The second `<a-point>` showcases some of the changes users can make to the annotation. 

**Anything you'd like to highlight / discuss:**

`<annotate>` functions very similarly to `<pic>` but after discussing with @damithc  we decided to split up the 2 classes to enable easier extension in the future. (E.g. adding the arrow implementation etc...)

Future improvements can be tracked at this issue #1911

Shoutout to @jovyntls for helping me with the reactivity of the points! 

**Testing instructions:**
Visit the Netlify Preview and head to `annotation` under images to test out the feature 

or 

1. inside a test markbind website run `markbind serve -d`
2. Copy the code snippet above and see if it works.
3. Play around with some of the other features

**Proposed commit message: (wrap lines at 72 characters)**

Adding a new Annotation Feature

Users want to annotate over images but have no way of doing so.

Implementing a `<annotate>` and `<a-point>` components to help users annotate over images.

This allows users to specify the image in `<annotate>`  and specify the points in `<a-point>`. 

Refer to issue #898 for initial suggestion.

---

**Checklist:** :ballot_box_with_check:
